sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/work/SignWorkAction.kt

```diff
@@ -52,7 +52,8 @@ abstract class SignWorkAction : WorkAction<SignWorkParameters> {
             oidcClient(parameters.oidcClient.get().build() as OidcClient)
         }.build()
 
-        val result = signer.sign(inputFile.toPath())
+        val results = signer.sign(*KeylessSigner.asDigests(inputFile.toPath()))
+        val result = results.get(0);
```

Comment:
```suggestion
        val result = signer.sign(*KeylessSigner.asDigests(inputFile.toPath())).first()
```

Reply:
However, currently we still create keylesssigner request per file, so it would be nice to batch signing requests automatically.

In other words, add stateful API like in
* `create signing session`: remember URLs, and other settings.
* `sign artifact`:  It should perform OIDC auth if missing, or it could just reuse already existing keys if they are still valid.
* `close signing session`: throw session keys away